portal/views/email.py

```diff
@@ -83,6 +83,39 @@ def verify_email(request, token):
     return HttpResponseRedirect(reverse_lazy('home'))
 
 
+def verify_email_new(request, token):
```

Comment:
You've also got a flag for cyclomatic complexity here, and again, it's probably the number of if statments, even though they're not nested.

In general terms, focus on what the output of this function should be, and outsource the rest of the logic to other functions.  You seem to have started doing that with your `verifications` var, but that gets abandoned further down. 

In JS this would be a good candidate for a `switch` statement, but Python doesn't have one.  Probably @duncanrussell-ocado or @CelineBoudier can advise better on actual syntax.